packages/lib-core/src/store/PluginStore.ts

```diff
@@ -195,20 +196,41 @@ export class PluginStore implements PluginConsumer, PluginManager {
       if (plugin.enabled !== enabled) {
         plugin.enabled = enabled;
         consoleLogger.info(`Plugin ${pluginName} will be ${enabled ? 'enabled' : 'disabled'}`);
-        updateExtensions = true;
+        update = true;
       }
     });
 
-    if (updateExtensions) {
-      this.extensions = Array.from(this.loadedPlugins.values()).reduce(
-        (acc, p) => (p.enabled ? [...acc, ...p.extensions] : acc),
-        [] as LoadedExtension[],
-      );
+    if (update) {
+      this.invokeListeners(PluginEventType.PluginInfoChanged);
+      this.updateExtensions();
+    }
+  }
+
+  updateExtensions() {
+    const prevExtensions = this.extensions;
 
-      this.invokeListeners([PluginEventType.PluginInfoChanged, PluginEventType.ExtensionsChanged]);
+    this.extensions = Array.from(this.loadedPlugins.values()).reduce(
+      (acc, p) => (p.enabled ? [...acc, ...p.extensions.filter(this.isExtensionInUse)] : acc),
+      [] as LoadedExtension[],
+    );
+
+    if (!_.isEqual(prevExtensions, this.extensions)) {
+      this.invokeListeners(PluginEventType.ExtensionsChanged);
     }
   }
 
+  /**
+   * Checks whether an extension is in use based on the values of required and disallowed feature flags
+   * @param {Extension} extension The extension to check for
+   * @returns {boolean} returns `true` if the extension is in use, and `false` if it is not in use
+   */
+  isExtensionInUse(extension: Extension) {
```

Comment:
```suggestion
  private isExtensionInUse(extension: Extension) {
```

Reply:
I'm OK with this change, but later on we'll want to unit test this method :smiley: 

Note that TypeScript `private` modifier is not the same as ES [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) which we currently don't use.